src/Datadog.Trace.ClrProfiler.Managed/Integrations/WebRequestIntegration.cs

```diff
@@ -83,7 +161,22 @@ public static object GetResponse(object webRequest, int opCode, int mdToken, lon
                 return callGetResponse(webRequest);
             }
 
-            using (var scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, request.Method, request.RequestUri, IntegrationId, out var tags))
+            var tracer = Tracer.Instance;
+
+            HttpTags tags;
+
+            var scope = ScopeFactory.GetActiveHttpScope(tracer);
+
+            if (scope == null)
+            {
+                scope = ScopeFactory.CreateOutboundHttpScope(tracer, request.Method, request.RequestUri, IntegrationId, out tags);
+            }
+            else
+            {
+                tags = scope.Span.Tags as HttpTags;
+            }
+
+            using (scope)
```

Comment:
In the case that there is an active scope (`ScopeFactory.GetActiveHttpScope(tracer)` returns non-null) this may dispose the parent `scope`. I'm not sure what behavior this will result in, but to keep the existing behavior, I'd suggest assigning the result of `ScopeFactory.GetActiveHttpScope(tracer)` to a different variable and use that variable in the null check above

Reply:
Very good catch

Reply:
I'm not sure I follow. I'm assuming `ScopeFactory.GetActiveHttpScope` will only return the scope created in GetRequestStream, in which case I want to dispose it. Are you worried about the case where we would have an active HTTP scope that is not the one from GetRequestStream? If that's the case I need to think of a way to uniquely identify the one we're interested in.

Reply:
Kevin and I chatted offline. I misunderstood the `GetRequestStream` implementation, but now I realized that method will open up a new `Scope` and the later call to this `GetResponse` method will close that `Scope`. We talked further and discussed a potentially better way to handle the `Scope` management which is to add the distributed tracing headers (same trace-id, new span-id) in `GetRequestStream` **but do not open a Scope**. Then when `GetResponse` is later called, check those headers and if they're present, use the span-Id in the header instead of a random ulong